spec/features/post_index_spec.rb

```diff
@@ -0,0 +1,54 @@
+require_relative '../rails_helper'
+
+RSpec.describe 'The page index page', type: :feature do
+  describe 'after logging in' do
+    before :each do
+      profile_picture = 'https://images.unsplash.com/photo-1546019170-f1f6e46039f5?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=928&q=80'
+      @user1 = User.create!(name: 'test', email: 'test@testmail.com', password: '123456', bio: 'test bio',
+                            photo: profile_picture, postsCounter: 1, confirmed_at: Time.now)
+
+      @post1 = @user1.posts.create!(title: 'Test Post', text: 'test post', commentCounter: 0, likesCounter: 2)
+      @post2 = @user1.posts.create!(title: 'Test Post 2', text: 'test post 2', commentCounter: 0, likesCounter: 2)
+      @post3 = @user1.posts.create!(title: 'Test Post 3', text: 'test post 3', commentCounter: 0, likesCounter: 2)
+
+      @comment1 = @post1.comments.create!(author: @user1, text: 'test comment')
+      @comment2 = @post1.comments.create!(author: @user1, text: 'test comment 2')
+
+      visit 'users/sign_in'
+      fill_in 'email', with: 'test@testmail.com'
+      fill_in 'password', with: '123456'
+      click_on 'Log in'
+    end
+
+    it "shows the user's username, profile picture and number of posts" do
+      visit "/users/#{@user1.id}/posts"
+      expect(page).to have_content 'test'
+      expect(page).to have_content 'Number of posts: 4'
+      expect(page).to have_css("img[src='https://images.unsplash.com/photo-1546019170-f1f6e46039f5?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=928&q=80']")
+    end
+
+    it "shows a post's title and some of it's body" do
+      visit "/users/#{@user1.id}/posts"
+      expect(page).to have_content 'Test Post'
+      expect(page).to have_content 'test post'
+    end
```

Comment:
- [ ] great job on every test you've done on this file. this are well written tests. However I notice you don't have a test to check whether you can see a section for pagination if there are more posts than fit on the view. Please kindly add this test to this file.

Reply:
Hi @MrBrN197, thank you for your feedback! I do not have a pagination functionality which is why I didn't add this test. I don't think it was required in previous tasks or was it? Thank you for your thoughts on this! I have implemented the other suggested changes.

Reply:
@Anjacodes  ok sure if that functionality wasn't required in the previous task. then you don't have to add a test for this. your other test are great.